static/js/datatables_config.js

```diff
@@ -0,0 +1,98 @@
+const gen_datatables_config = (overwrites) => {
+  const column_selector = (idx, data, node) => {
+    // https://datatables.net/forums/discussion/42192/exporting-data-with-buttons-and-responsive-extensions-controlled-by-column-visibility
+    if ($(node).hasClass('noVis')) {
+      return false;
+    }
+    const table = $(node).closest('table');
+    return table.length === 0 ? false : table.DataTable().column(idx).visible();
```

Comment:
In what case can this be null?

Reply:
When the responsive/colvis plugin hides an column it does it by either adding to it the attribute noVis or by physically detaching it from the table - this is why it didn't worked before.

When the element is detached we won't find a parent table.

Reply:
That's kinda stupid (why does it do one or the other?!) but fine, just add a comment explaining it